test/pg.ddl.unit.js

```diff
@@ -0,0 +1,788 @@
+          },
+        ],
+        'Schema'
+      ),
+      `\
+ALTER TABLE "Schema" ADD CONSTRAINT "akSchemaunique1" UNIQUE ("unique1");
+ALTER TABLE "Schema" ADD CONSTRAINT "akSchemaunique3" \
+UNIQUE ("unique2", "unique3");`
+    )
+  );
+
+  unitTests.testSync('pg.ddl generateId', test => {
+    test.strictSame(ddl.generateId('Local', 2), '"Id" bigserial');
+    test.strictSame(ddl.generateId('Global', 2), '"Id" bigint');
+  });
+
+  unitTests.testSync('pg.ddl getResolvalbeLinks', test => {
+    test.strictSame(
+      ddl.getResolvableLinks(
+        'Schema1',
+        new Map(),
+        [
+          {
+            name: 'field',
+            property: { category: 'Schema2' },
+            destination: 'Schema2',
+          },
+          {
+            name: 'field',
+            property: { category: 'Schema1' },
+            destination: 'Schema1',
+          },
+          {
+            name: 'field',
+            property: { category: 'Schema4' },
+            destination: 'Schema4',
+          },
+        ],
+        ['Schema1', 'Schema2']
+      ),
+      [
+        [
+          {
+            name: 'field',
+            property: { category: 'Schema4' },
+            destination: 'Schema4',
+          },
+        ],
+        [
+          {
+            from: 'Schema1',
+            to: 'Schema2',
+            name: 'field',
+            link: { category: 'Schema2' },
+            destination: 'Schema2',
+          },
+          {
+            from: 'Schema1',
+            to: 'Schema1',
+            name: 'field',
+            link: { category: 'Schema1' },
+            destination: 'Schema1',
+          },
+        ],
+      ]
+    );
+
+    test.strictSame(
+      ddl.getResolvableLinks(
+        'Schema1',
+        new Map([
+          ['Schema1', [
+            {
+              from: 'Schema1',
+              to: 'Schema2',
+              name: 'field',
+              link: { category: 'Schema2' },
+              destination: 'Schema2',
+            },
+          ]],
+        ]),
+        [],
+        ['Schema1', 'Schema2']
+      ),
+      [
+        [],
+        [
+          {
+            from: 'Schema1',
+            to: 'Schema2',
+            name: 'field',
+            link: { category: 'Schema2' },
+            destination: 'Schema2',
+          },
+        ],
+      ]
+    );
+  });
+
+  unitTests.testSync('pg.ddl generateTable', test =>
+    test.strictSame(
+      ddl.generateTable(
+        'LocalCategory1',
+        ms.categories.get('LocalCategory1').definition,
+        'Local',
+        new Map([['Nomen', 'text']]),
+        new Map(), ['Table'], ms.categories
+      ),
+      {
+        sql: `
+-- Category: LocalCategory1 ----------------------------------------------------
+
+CREATE TABLE "LocalCategory1" (
+  "Id"    bigserial,
+  "field" text
+);
+
+ALTER TABLE "LocalCategory1" ADD CONSTRAINT "pkLocalCategory1Id" \
+PRIMARY KEY ("Id");`,
+        unresolved: [],
```

Comment:
You should also add a test, to ensure that unresolved links are returned from this function.